Source/ASCollectionView.mm

```diff
@@ -970,13 +972,11 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
 
   BOOL shouldDequeueExternally = _asyncDataSourceFlags.interopViewForSupplementaryElement && (_asyncDataSourceFlags.interopAlwaysDequeue || node.shouldUseUIKitCell);
   if (shouldDequeueExternally) {
+    // This codepath is used for both IGListKit mode, and app-level UICollectionView interop.
     view = [(id<ASCollectionDataSourceInterop>)_asyncDataSource collectionView:collectionView viewForSupplementaryElementOfKind:kind atIndexPath:indexPath];
   } else {
-    view = [self dequeueReusableSupplementaryViewOfKind:kind withReuseIdentifier:kReuseIdentifier forIndexPath:indexPath];
-  }
-
-  if (!node.shouldUseUIKitCell) {
     ASDisplayNodeAssert(node != nil, @"Supplementary node should exist.  Kind = %@, indexPath = %@, collectionDataSource = %@", kind, indexPath, self);
```

Comment:
@Adlai-Holler  For some reason, I was hitting this assertion in the passthrough cases I was testing. I agree this probably shouldn't be the case, but I think it could be an artifact of issues with the supplementary node system (or potentially the test case).

I'll make a note for myself to investigate this assertion in more detail and will send another PR if I can get it re-added.